### ISSUE TITLE
addressToLine misbehaves in master

### ISSUE THREAD

**Issue by ilejn** [ClickHouse/ClickHouse#31639]
Note filenames in the backtrace

```
tail -10  clickhouse-server.err.log
5. DB::TCPHandler::receiveHello() @ 0x131f2503 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
6. DB::TCPHandler::runImpl() @ 0x131ebd82 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
7. DB::TCPHandler::run() @ 0x13200499 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
8. Poco::Net::TCPServerConnection::start() @ 0x15e8358f in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
9. Poco::Net::TCPServerDispatcher::run() @ 0x15e85981 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
10. Poco::PooledThread::run() @ 0x15f9c1c9 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
11. Poco::ThreadImpl::runnableEntry(void*) @ 0x15f99900 in /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug
12. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
13. __clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
 (version 21.12.1.8761 (official build))

```
Playing with 00974_query_profile.sql I see the same

```
select addressToLine(toUInt64(293109982));

SELECT addressToLine(toUInt64(293109982))

Query id: 7d7353c9-f4f7-4f81-8d96-3e9e2b6defc6

┌─addressToLine(toUInt64(293109982))───────────────────────────────────────┐
│ /usr/lib/debug/.build-id/03/836aa56cfad660466eb42a7973e7053e525ccb.debug │
└──────────────────────────────────────────────────────────────────────────┘


```

Unfortunately, the test does not treat this as failure because condition 'symbol LIKE '%Source%';' is true for some records

It might be that the issue caused by something strange at my box. 
Please, let me know if the issue is reproducible (or not).

**Comment by ilejn** [comment]
dpkg -l | grep clickh
```
ii  clickhouse-client                    21.12.1.8761                                                   all          Client binary for ClickHouse
ii  clickhouse-common-static             21.12.1.8761                                                   amd64        Common files for ClickHouse
ii  clickhouse-common-static-dbg         21.12.1.8761                                                   amd64        debugging symbols for clickhouse-common-static
ii  clickhouse-server                    21.12.1.8761                                                   all          Server binary for ClickHouse

```

lsb_release -a
```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
```

**Comment by ilejn** [comment]
FunctionAddressToLine::impl returned 0 and FunctionAddressToLine::impl returned object->name

            if (dwarf_it->second.findAddress(addr - uintptr_t(object->address_begin), location, Dwarf::LocationInfoMode::FAST, frames))
            {
                const char * arena_begin = nullptr;
                WriteBufferFromArena out(cache.arena, arena_begin);

                writeString(location.file.toString(), out);
                writeChar(':', out);
                writeIntText(location.line, out);

                return out.finish();
            }
            else
            {
                return object->name;

**Comment by azat** [comment]
This should be fixed by #40873
